### PR TITLE
Move coin type to AccountManager

### LINKED ISSUE #1217: [Task]: Move coin type to account manager
### Task description

Move coin type to account manager, so it's clearer that all accounts use the same coin type

### Requirements

-

### Acceptance criteria

-

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Move coin type to AccountManager so it's clearer, that all accounts share the same

## Links to any relevant issues

Fixes #1217 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Tests

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/python/native/example/1-create-account.py [hunk 1]
```diff
@@ -9,6 +9,6 @@
 secret_manager = ("flame fever pig forward exact dash body idea link scrub tennis minute " +
           "surge unaware prosper over waste kitten ceiling human knife arch situate civil")
 
-wallet = IotaWallet('./alice-database', client_options, secret_manager)
+wallet = IotaWallet('./alice-database', client_options, coin_type: 4219, secret_manager)
```
Comment (thibault-martinez) [inline bindings/python/native/example/1-create-account.py:12]: Don't we have a constant ?

Comment (Thoralf-M) [reply bindings/python/native/example/1-create-account.py:12]: Not in Python

File: tests/account_manager.rs [hunk 2]
```diff
@@ -98,3 +104,124 @@ async fn different_seed() -> Result<()> {
...
+    // mnemonic without balance
+    let secret_manager = MnemonicSecretManager::try_from_mnemonic(
+        "inhale gorilla deny three celery song category owner lottery rent author wealth penalty crawl hobby obtain glad warm early rain clutch slab august bleak",
+    )?;
+
+    let manager = AccountManager::builder()
+        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
+        .with_client_options(client_options)
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .with_coin_type(SHIMMER_COIN_TYPE)
```
Comment (thibault-martinez) [inline tests/account_manager.rs]: Is this on purpose ?

Comment (Thoralf-M) [reply tests/account_manager.rs]: No

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**